### PR TITLE
fix meta#493: text disappears when focusing textfield

### PR DESCRIPTION
**Pull request by gtalha07**

### CODE REVIEW

File: app/lib/features/chat/widgets/custom_input.dart [hunk 1]
```diff
@@ -615,13 +611,8 @@ class __ChatInputState extends ConsumerState<_ChatInput> {
         const Spacer(),
         GestureDetector(
           onTap: () async {
-            final convo = await ref.read(chatProvider(widget.roomId).future);
-            await convo?.saveMsgDraft(textController.text, null, 'new', null);
```
Comment (gtalha07) [inline app/lib/features/chat/widgets/custom_input.dart:619]: This is not needed as listener updates controller and calls ```saveDraft()```

File: app/lib/features/chat/widgets/custom_input.dart [hunk 2]
```diff
@@ -615,13 +611,8 @@ class __ChatInputState extends ConsumerState<_ChatInput> {
...
             inputNotifier.unsetSelectedMessage();
-            // frame delay to keep focus connected with keyboard.
-            WidgetsBinding.instance.addPostFrameCallback((_) {
```
Comment (gtalha07) [inline app/lib/features/chat/widgets/custom_input.dart:622]: Had no effect on focus...

File: app/lib/features/chat/widgets/custom_input.dart [hunk 3]
```diff
@@ -217,19 +218,14 @@ class __ChatInputState extends ConsumerState<_ChatInput> {
...
-    super.didChangeDependencies();
-    _setController();
-  }
-
   void _setController() {
+    // putting constant colors here as context isn't accessible in initState()
     final triggerStyles = {
       '@': TextStyle(
-        color: Theme.of(context).colorScheme.onSecondary,
+        color: Colors.white,
```
Comment (gnunicorn) [inline app/lib/features/chat/widgets/custom_input.dart:225]: why?

File: app/lib/features/chat/widgets/custom_input.dart [hunk 4]
```diff
@@ -217,19 +218,14 @@ class __ChatInputState extends ConsumerState<_ChatInput> {
...
         height: 0.5,
         background: Paint()
-          ..color = Theme.of(context).colorScheme.secondary
+          ..color = const Color(0xFF74A64D)
```
Comment (gnunicorn) [inline app/lib/features/chat/widgets/custom_input.dart:228]: oh, noes... static color ... I see the almighty looking down disapprovingly to us ... I can feel his pain and sorrow in this sight ...

**Review by gnunicorn — APPROVED** [review]
Tests, nice! Love it!

Just a minor thing about the static colors being added.

File: app/lib/features/chat/widgets/custom_input.dart (hunk 4 above)
Comment (gtalha07) [reply app/lib/features/chat/widgets/custom_input.dart:228]: Sorry, but it's just how the framework works. We can't have ```BuildContext``` before widget build and we need to set it once only in ```initState()```.

**Pull request merged**